src/js/jsx/search/SearchBar.jsx

```diff
@@ -250,41 +303,232 @@ define(function (require, exports, module) {
 
             var recentDocLabel = {
                 id: "recent-doc_header",
-                title: "Recent Documents",
+                title: HEADERS.RECENT_DOCS,
                 type: "header"
             };
 
-            recentDocMap.unshift(recentDocLabel);
+            return recentDocMap.unshift(recentDocLabel);
+        },
+
+        /**
+         * Make list of search category dropdown options based on header
+         * 
+         * @param {string} header, either "layer" or "document"
+         * @return {Array.<object>}
+         */
+        _getFilterOptions: function (header) {
+            if (header !== "layer" && header !== "document") {
+                return;
+            }
+            var categoryList = header === "document" ? ["current", "recent"] : layerLib.layerKinds,
+                categories = Immutable.fromJS(categoryList).filterNot(function (kind) {
+                                return (kind === layerLib.layerKinds.ANY || kind === layerLib.layerKinds.GROUPEND ||
+                                    kind === layerLib.layerKinds["3D"] || kind === layerLib.layerKinds.VIDEO);
+                            }).toList(),
+                                   
+                filters = categories.map(function (kind) {
+                    var title,
+                        idType;
+
+                    if (header === "layer") {
+                        idType = CATEGORIES.LAYER_KIND[kind];
+                        title = CATEGORIES.LAYER_KIND[kind];
+                    } else {
+                        idType = CATEGORIES[kind.toUpperCase()];
```

Comment:
Is there some way to get rid of all this case manipulation? Can the keys just all be made upper case, for example?